encodings/fastlanes/src/for/compress.rs

```diff
@@ -116,10 +116,14 @@ pub fn decompress(array: FoRArray) -> VortexResult<PrimitiveArray> {
                 .as_primitive()
                 .typed_value::<$T>()
                 .ok_or_else(|| vortex_err!("expected reference to be non-null"))?;
-            PrimitiveArray::from_vec(
-                decompress_primitive(encoded.into_maybe_null_slice::<$T>(), min, shift),
-                validity,
-            )
+            if min == 0 && shift == 0 {
+                encoded
```

Comment:
do we need to cast? (real question)

Reply:
oh no, I mean cast `encoded`... is it the right dtype/ptype? since FoR converts to unsigned

Reply:
The cast happens above when we extract the encoded array